manage_breast_screening/utils/date_formatting.py

```diff
@@ -0,0 +1,94 @@
+"""
+Format and dates and times in a way consistent with
+the NHS service manual's content guide:
+https://service-manual.nhs.uk/content/numbers-measurements-dates-time
+"""
+
+from datetime import date, datetime
+
+
+def format_date(value):
+    """
+    Format a date
+
+    >>> from datetime import date
+    >>> format_date(date(2025, 1, 1))
+    '1 January 2025'
+    """
+    return value.strftime("%-d %B %Y")
+
+
+def format_relative_date(value: datetime | date):
+    """
+    Format a date relative to today.
+    """
+    if isinstance(value, datetime):
+        value = value.date()
+
+    today = date.today()
+    if value == today:
+        return "today"
+
+    days = (value - today).days
+
+    if -6 <= days <= -2:
+        return f"{abs(days)} days ago"
+    elif days == -1:
+        return "yesterday"
+    elif days == 1:
+        return "tomorrow"
+    elif 2 <= days <= 6:
+        return f"in {days} days"
+    else:
+        return format_date(value)
```

Comment:
In mine, this falls back to the dayjs relative date - which might be n days, weeks, months or years. Then with an optional suffix.

Reply:
Pushed up another iteration of this, but I've left out the weeks, months, years logic that is currently performed by dayjs. As discussed on slack it needs a bit more thought than I'm prepared to give it right now, and I didn't find a library method that does exactly the same thing out of the box.